src/lib/structures/typed.mli

```diff
@@ -285,6 +285,11 @@ and 'a tdecl =
   | TExit of Loc.t
   (** Exits the solver. *)
 
+  | TOptimize of Loc.t * 'a atterm * bool
+  (** Optimization declaration.
+      [TOptimize (loc, obj, to_max] declares an objective function [obj]. The
+      flag [to_max] determines if we try to maximize of minimize [obj]. *)
```

Comment:
Don't want to quibble over it, but `to_max` is used as `is_max` everywhere else the code

Reply:
Actually I used both. I will put `to_max` everywhere.